wallet/wallet.go

```diff
@@ -3706,6 +3713,79 @@ func (w *Wallet) LockedOutpoints() []dcrdtypes.TransactionInput {
 	return locked
 }
 
+func (w *Wallet) acctLockedOutpoints(ctx context.Context, acctName string) ([]dcrdtypes.TransactionInput, error) {
+	w.lockedOutpointMu.Lock()
+	locked := make([]wire.OutPoint, len(w.lockedOutpoints))
+	i := 0
+	for op := range w.lockedOutpoints {
+		locked[i] = op
+		i++
+	}
+	w.lockedOutpointMu.Unlock()
+
+	acctLocked := make([]dcrdtypes.TransactionInput, 0, len(w.lockedOutpoints))
+	err := walletdb.View(ctx, w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
+
+		defaultAccountName, err := w.manager.AccountName(
+			addrmgrNs, udb.DefaultAccountNum)
+		if err != nil {
+			return err
+		}
+
+		// Outpoints are not validated before they're locked, so
+		// invalid outpoints may be locked. Simply ignore.
+		for _, op := range locked {
+			details, err := w.txStore.TxDetails(txmgrNs, &op.Hash)
+			if err != nil {
+				if errors.Is(err, errors.NotExist) {
+					// invalid txid
+					continue
+				}
+				return err
+			}
+			if int(op.Index) >= len(details.MsgTx.TxOut) {
+				// valid txid, invalid vout
+				continue
+			}
+			output := details.MsgTx.TxOut[op.Index]
+			// Lookup the associated account for the output.  Use the
+			// default account name in case there is no associated account
+			// for some reason, although this should never happen.
+			//
+			// This will be unnecessary once transactions and outputs are
+			// grouped under the associated account in the db.
+			opAcct := defaultAccountName
+			_, addrs, _, err := txscript.ExtractPkScriptAddrs(
+				0, output.PkScript, w.chainParams)
```

Comment:
I imagine `output.Version` would be used instead of 0, but there seems to be inconsistent use of `ExtractPkScriptAddrs` in this respect in this file so I'm not sure.